deepchem/rl/a3c.py

```diff
@@ -21,10 +21,11 @@ def __init__(self, value_weight, entropy_weight, **kwargs):
     self.entropy_weight = entropy_weight
 
   def create_tensor(self, **kwargs):
-    reward, action, prob, value = [layer.out_tensor for layer in self.in_layers]
-    log_prob = tf.log(prob)
-    policy_loss = -tf.reduce_sum(
-        (reward - value) * tf.reduce_sum(action * log_prob))
+    reward, action, prob, value, advantages = [
+        layer.out_tensor for layer in self.in_layers
+    ]
+    log_prob = tf.log(prob + 0.0001)
+    policy_loss = -tf.reduce_sum(advantages * tf.reduce_sum(action * log_prob))
```

Comment:
That isn't necessarily the correct equation to use.  They say the update "can be seen as" an estimate of that function.  However, the actual loss function they use appears two paragraphs later.  It just depends on `R-V` without that extra term.

Also see https://arxiv.org/pdf/1506.02438.pdf for a much more detailed discussion of advantage estimation.  They present both forms (eq. 17 and 18) as limiting cases of a continuous family of advantage estimates (eq. 16).  The one I implemented (`lambda=1`) is unbiased but has larger variance.  The one you used (`lambda=0`) reduces the variance but introduces bias.  In their experiments, they get the best results for values in the range of about 0.96, so much closer to 1 than 0.  We could consider adding `lambda` as a user selectable hyperparameter.